### DISCUSSION TITLE
Updating, Adding and Deleting users oAuth providers

### DISCUSSION THREAD

**Discussion by xzilja** [supabase/supabase#9149] [Questions] [answered]
Hello 👋 

Was curious if it is possible to update oAuth provider of existing user i.e. if they signed up with twitter, but now want to change it to google for example?

In addition to above, I know this is possible in firebase, but not sure if currently available in supabase: Can user link more than one oAuth provider to their account? i.e. If I am already signed up with twitter, but would also like to add discord as my login method for redundancy reasons. Naturally if this is available, can users then remove these extra providers?

**Comment by kangmingtay** [comment, marked as answer]
Hey @IljaDaderko, supabase auth currently doesn't support manual linking of oauth provider accounts. However, we automatically link 2 oauth accounts if the oauth accounts (twitter and google in this case) have the same email. 

You can link more than 1 oauth provider to the same account via automatic linking as long as the other oauth provider accounts have the same email address as the account you are trying to link to.

This means is that you don't have to call any additional APIs / methods on your end to link accounts since the automatic linking happens natively in supabase auth.
